engine/lib/phases/phase_drop_return_break_continue.ml

```diff
@@ -0,0 +1,134 @@
+(* This phase removes `return`s in exit position. Inside loops,
```

Comment:
We chatted about this: we should make this phase typed, but since the visitor by design doesn't need to visit every nodes, rejecting within the rewrite is difficult.
Let's then just add a treatment after the visitor that takes care of rejecting (with an assertion failure) any return/break/continue

Reply:
Also, we want `Loop` in AST to carry a ``([ `BreakOnly | `BreakOrReturn ] * F.fold_like_loop) option``, and we want this phase to enable `fold_like_loop`.